pinnable.go

```diff
@@ -0,0 +1,38 @@
+package ebpf
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/cilium/ebpf/internal"
+)
+
+func pin(currentPath, newPath string, fd *internal.FD) error {
+	if newPath == "" {
+		return fmt.Errorf("pinned path cannot be empty")
+	}
+	if currentPath != "" {
+		if currentPath == newPath {
+			return nil
+		}
+		if err := os.Rename(currentPath, newPath); err != nil {
+			if !os.IsNotExist(err) {
```

Comment:
I would put this at the root indentation level at the end of the function:

```go
if err := os.Rename(currentPath, newPath); err != nil && os.IsNotExist(err) {
  return internal.BPFObjPin(newPath, fd)
} else {
  return fmt.Errorf("unable to move object pin to new path %v: %w", newPath, err)
}
```

This removes a few layers of indentation and makes this easier to follow. Note: also edited the error message.

Reply:
Channeling `go vet`: you can drop the `else` as well.

Reply:
@ti-mo I don't think that code block works.

Reply:
@aditighag Sorry, why would this not work?

Building on what @lmb suggested, this would be ideal:

```go
func pin( ... ) error {

...

	err := os.Rename(currentPath, newPath)
	if err != nil && os.IsNotExist(err) {
		return internal.BPFObjPin(newPath, fd)
	}
	
	return fmt.Errorf("unable to move object pin to new path %v: %w", newPath, err)
}
```

Reply:
>Sorry, why would this not work?

@ti-mo Did you try the multiple pin calls case?

Reply:
>Sorry, why would this not work?

Unless I'm missing something, your suggested code block doesn't cover the happy path (i.e. when `os.Rename succeeds`). Can you try running the `TestMapPinMultiple` unit test case?

Reply:
Whoops, lost track of the goal. :sweat_smile: I like the current implementation! :+1: